perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/BBMRIResources.java

```diff
@@ -0,0 +1,317 @@
+					"If you approve the application, these resources will be skipped.", "", "", "", true, app.getId());
+		}
+	}
+
+	/**
+	 * Gets Resources present in the system as a map of the Resource ID to the group representing particular resource.
+	 *
+	 * @return Map of String to Group, where key is the ID of the resource and Group is the representation
+	 */
+	private Map<String, Group> getPerunResourceIdToGroupMap(PerunSession session, Application app, Perun perun)
+		throws PrivilegeException, RegistrarException, VoNotExistsException, WrongAttributeAssignmentException,
+		AttributeNotExistsException, GroupNotExistsException
+	{
+		// get root group for resources hierarchy
+		Group resourceOriginGroup = getResourceOriginGroup(session, app, perun);
+		// get name of the attribute (stored in RESOURCE_ID_ATTR_NAME attribute) containing ID of the resource
+		String resourceIdAttributeName = getResourceIdAttributeName(session, app, perun);
+
+		// get map of ResourceID -> group (representing resource)
+		return getResourceIDsToGroupsMap(session, perun, resourceOriginGroup, resourceIdAttributeName);
+	}
+
+	/**
+	 * Gets name of the attribute containing Resource IDs. This attribute should be used at the groups
+	 * identifying resources.
+	 *
+	 * @return name of the attribute where ID of the resource is stored, null if value is not set
+	 */
+	private String getResourceIdAttributeName(PerunSession session, Application app, Perun perun)
+		throws WrongAttributeAssignmentException, AttributeNotExistsException, GroupNotExistsException,
+		PrivilegeException
+	{
+		return perun.getAttributesManager()
+			.getAttribute(session, app.getGroup(), RESOURCE_ID_ATTR_NAME)
+			.valueAsString();
+	}
+
+	/**
+	 * Gets root group, under which subgroups representing resources are placed.
+	 *
+	 * @return resource IDs set
+	 */
+	private Group getResourceOriginGroup(PerunSession session, Application app, Perun perun)
+		throws PrivilegeException, RegistrarException, VoNotExistsException, GroupNotExistsException
+	{
+		try {
+			if (perun.getAttributesManager()
+				.getAttribute(session, app.getGroup(), RESOURCE_ORIGIN_ENABLED_ATTR_NAME)
+				.valueAsBoolean()
+			) {
+				try {
+					String resourceOriginGroupName = this.getResourceOriginGroupNameFromApplication(session, app);
+					return perun.getGroupsManager().getGroupByName(session, app.getVo(), resourceOriginGroupName);
+				} catch (GroupNotExistsException e) {
+					throw new InternalErrorException("Target group does not exist");
+				}
+			}
+		} catch (AttributeNotExistsException | WrongAttributeAssignmentException ex) {
+			//OK, we consider it as disabled, try manually configured resource root group
+		}
+
+		try {
+			try {
+				String resourceOriginGroupName = perun.getAttributesManager()
+					.getAttribute(session, app.getGroup(), RESOURCES_ROOT_GROUP_ATTR_NAME)
+					.valueAsString();
+				return perun.getGroupsManager().getGroupByName(session, app.getVo(), resourceOriginGroupName);
```

Comment:
I suggest to use Bl instead of Entry layer for this kind of operations (getting groups, attributes, subgroups), since for example in this case you sort of redirect the root of resources to a completely different group, where approving person might not be the group manager and hence might not be able to retrieve relevant data. Same goes for the adding member to such group (if it will be outside of the sub-group tree).

Reply:
Thanks for the suggestion, I have applied BL for every call :)